apiserver/common/networkingcommon/linklayer.go

```diff
@@ -178,16 +203,23 @@ func (o *MachineLinkLayerOp) AssertAliveOp() txn.Op {
 	return o.machine.AssertAliveOp()
 }
 
-// MarkProcessed indicates that the input (known) device was present in the
+// MarkDevProcessed indicates that the input (known) device was present in the
 // incoming data and its updates have been handled by the build step.
-func (o *MachineLinkLayerOp) MarkProcessed(dev LinkLayerDevice) {
-	o.processed.Add(dev.MACAddress())
+func (o *MachineLinkLayerOp) MarkDevProcessed(dev LinkLayerDevice) {
+	o.processedDevs.Add(dev.MACAddress())
+}
+
+// IsDevProcessed returns a boolean indicating whether the input incoming
+// device matches a known device that was marked as processed by the method
+// above.
+func (o *MachineLinkLayerOp) IsDevProcessed(dev network.InterfaceInfo) bool {
+	return o.processedDevs.Contains(dev.MACAddress)
 }
 
-// IsProcessed returns a boolean indicating whether the input incoming device
-// matches a known device that was marked as processed by the method above.
-func (o *MachineLinkLayerOp) IsProcessed(dev network.InterfaceInfo) bool {
-	return o.processed.Contains(dev.MACAddress)
+// MarkAddrProcessed indicates that the input (known) IP address was present in
+// the incoming data and its updates have been handled by the build step.
+func (o *MachineLinkLayerOp) MarkAddrProcessed(ipAddress string) {
+	o.processedAddrs.Add(ipAddress)
```

Comment:
I hope we never ever have to deal with a dual-NIC machine where both NICs have the same address on physically isolated networks...

Reply:
Yes indeed. I will fix this now, by grouping these up by NIC.